client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -345,4 +358,42 @@ public void close() {
             this.connectionPool.close();
         }
     }
+
+    public CompletableFuture<Long> getDistanceBetweenTwoStreamCuts(Stream stream, StreamCut fromStreamCut, StreamCut toStreamCut) {
+        final CompletableFuture<StreamSegmentSuccessors> unread;
+        final Map<Segment, Long> endPositions;
+        if (toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            CompletableFuture<StreamInfo> streamInfo = fetchStreamInfo(stream.getScope(), stream.getStreamName());
+            StreamCut endSC = streamInfo.join().getTailStreamCut();
+            unread = controller.getSegments(fromStreamCut, endSC);
+            endPositions = endSC.asImpl().getPositions();
+        } else {
+            unread = controller.getSegments(fromStreamCut, toStreamCut);
+            endPositions = toStreamCut.asImpl().getPositions();
+        }
+        return unread.thenCompose(unreadVal -> {
+            DelegationTokenProvider tokenProvider = DelegationTokenProviderFactory
+                    .create(controller, stream.getScope(), stream.getStreamName(), AccessOperation.READ);
+            return Futures.allOfWithResults(unreadVal.getSegments().stream().map(s -> {
+                if (endPositions.containsKey(s)) {
+                    return CompletableFuture.completedFuture(endPositions.get(s));
+                } else {
+                    SegmentMetadataClient metadataClient = metaFactory.createSegmentMetadataClient(s, tokenProvider);
+                    CompletableFuture<Long> result = metadataClient.fetchCurrentSegmentLength();
+                    result.whenComplete((r, e) -> metadataClient.close());
+                    return result;
+                }
+            }).collect(Collectors.toList()));
+        }).thenApply(sizes -> {
+            long totalLength = 0;
+            for (long bytesRemaining : sizes) {
+                totalLength += bytesRemaining;
+            }
+            for (long bytesRead : fromStreamCut.asImpl().getPositions().values()) {
+                totalLength -= bytesRead;
+            }
+            log.debug("Remaining bytes from position: {} to position: {} is {}", fromStreamCut, toStreamCut, totalLength);
```

Comment:
This isn't properly accounting for toStreamCut. It may be that the toStreamCut ends before the end of the segment.